### PR TITLE
Linter and a bunch of fixes

### PR DESCRIPTION
**Pull request by marmarta**
Support for pylint and mypy , keynav, a bunch of fixes for issues found by mypy and pylint.

### CODE REVIEW

**Comment by marmarta** [comment]
@marmarek , what is needed in this repo to make it run CI?

**Comment by marmarek** [comment]
PipelineRetry

**Comment by marmarek** [comment]
PipelineRetry

**Comment by marmarta** [comment]
PipelineRetry

**Comment by marmarta** [comment]
@marmarek , I'm adding tests now, but otherwise this is ready for a review.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,104 @@
...
+The menu consists of three panes: Applications, Favorites and System Tools.
+
+### Applications pane
+
+The Application pane contains a list of all qubes found in user's system,
+sorted into App qubes (with no special role), Templates (both "normal"
+templates and Disposable VM templates) and Service qubes (such as `sys-usb` 
+or `sys-net`). Running qubes' names are bolded.
+
+![](menu1.png)
```
Comment (marmarek) [inline README.md:18]: Please place those files in some subdir.

File: debian/control [hunk 2]
```diff
@@ -11,24 +11,20 @@ Build-Depends:
...
 Standards-Version: 3.9.5
 Homepage: https://www.qubes-os.org/
-#Vcs-Git: git://github.com/QubesOS/qubes-core-admin-client.git
-#Vcs-Browser: https://github.com/QubesOS/qubes-core-admin-client
 X-Python3-Version: >= 3.5
 
 Package: qubes-desktop-linux-menu
 Architecture: any
 Depends:
- python3-qui,
```
Comment (marmarek) [inline debian/control:21]: This package should depend on `python3-qubes-menu`

File: debian/rules [hunk 3]
```diff
@@ -15,8 +15,8 @@ override_dh_auto_build:
 
 override_dh_auto_install:
 	dh_auto_install
-	make install DESTDIR=debian/qubes-desktop-linux-manager
-	mv debian/python3-qui/usr/bin/* debian/qubes-desktop-linux-manager/usr/bin/
+	make install DESTDIR=debian/qubes-desktop-linux-menu/usr/bin/
```
Comment (marmarek) [inline debian/rules:18]: `DESTDIR` should not include `/usr/bin`

**Review by marmarek — CHANGES_REQUESTED** [review]
Few more comments for parts unchanged in this PR:
 - `Makefile`: intsall icons into `hicolor` icon theme
 - `Makefile.builder`: `RPM_BUILD_DEFINES` should not be necessary
 - rpm spec: several dependencies are not needed here
 - `debian/copyright`: update author
 - `qubes-menu.glade~` should not be here

File: qubes_menu/desktop_file_manager.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        self.vm: Optional[qubesadmin.vm.QubesVM] = None
+        self.entry_name: Optional[str] = None
+        self.exec: List[str] = []
+        self.disposable: bool = False
+        self.categories = []
+        self.entries: List = []
+
+    def load_data(self, entry):
+        """Fill own data with information from xdg.DesktopEntry provided."""
+        vm_name = entry.get('X-Qubes-VmName') or None
```
[the remark this answers is not in the document]
Comment (marmarta) [reply qubes_menu/desktop_file_manager.py:60]: No, it returns an empty string

**Comment by marmarta** [comment]
Changes made. Will add more tests in coming days, but otherwise this is ready to merge (if the build succeeds).

**Review by marmarek — APPROVED** [review]

**Pull request merged**